packages/common/src/ui/layout/tables/utils/ColumnSetBuilder.ts

```diff
@@ -0,0 +1,161 @@
+import { getCheckboxSelectionColumn } from './../columns/CheckboxSelectionColumn';
+import { ColumnAlign } from '@openmsupply-client/common/src/ui/layout/tables/columns/types';
+import { ColumnFormat } from './../columns/types';
+import { DomainObject } from './../../../../types';
+import { ColumnDefinition } from '../columns/types';
+
+const createColumn = <T extends DomainObject>(
+  column: ColumnDefinition<T>
+): ColumnDefinition<T> => {
+  return column;
+};
+
+export type ColumnKey =
+  | 'invoiceNumber'
+  | 'type'
+  | 'status'
+  | 'entered'
+  | 'confirmed'
+  | 'total'
+  | 'comment'
+  | 'selection'
+  | 'name'
+  | 'code'
+  | 'packSize'
+  | 'quantity';
+
+const getColumnLookup = <T extends DomainObject>(): Record<
```

Comment:
I tried to get the typings of this nicer. I'm sure it must be possible but i'll need to revisit unless you have some ideas.

I wanted to constrain the `columnKey` passed to `addColumn` to be only those in the set of `keyof T` also. I couldn't get this to work. If I had, then I would have also possibly had to figure out a solution for columns which are not of `keyof T`, but are legal, such as `selection` 🤔

Reply:
yeah - no great ideas sorry. I think we should merge this and revisit